lib/communicator_qmp.cpp

```diff
@@ -166,6 +166,9 @@ void Communicator::comm_init(int ndim, const int *dims, QudaCommsMap rank_from_c
               grid_size, QMP_comm_get_number_of_nodes(QMP_COMM_HANDLE));
   }
 
+  // defer handling MPI errors to QMP
```

Comment:
I'm a bit confused, in `lib/communicator_mpi.cpp` an identical function call defers handling MPI errors to QUDA, why does this one defer it to QMP? Is this just a statement of "who called MPI_Init"?

Reply:
I don't believe we have the capability to defer the QMP error handling to QUDA, though that would be desirable.  It's something that I will look into adding to QMP and then propagate enabling it into QUDA.

The motivation in the first place for this is to allow us to do post-error analysis, e.g., print a stack trace if `QUDA_BACKWARDS` is enabled, or anything else we might want to do if an MPI error is thrown.  I found this very helpful when encountering fatal MPI errors.